src/specific/s_filesystem.c

```diff
@@ -35,3 +58,71 @@ void S_File_CreateDirectory(const char *path)
     mkdir(path, 0664);
 #endif
 }
+
+// r must have strlen(path) + 2 bytes
+bool S_File_CasePath(char const *path, char *r)
+{
+    size_t length = strlen(path);
+    char *p = Memory_Alloc(length + 1);
+    strcpy(p, path);
+    size_t rl = 0;
+
+    DIR *d;
+    if (p[0] == '/') {
+        d = opendir("/");
+        p = p + 1;
+    } else {
+        d = opendir(".");
+        r[0] = '.';
```

Comment:
This is where we make the path start with `./`

Reply:
I'll have to do some more testing. If I remove that, it bugs the directory traversal process.

Reply:
I updated all the names because the `c`, `d`, `p`, etc were not clear at all.

The problem seems to be removing `./` causes `case_path[case_len] = '/';` to need to be moved near `path_piece = S_File_Strsep(&path_copy, "/");`, This change causes `path_dir = opendir(case_path);` to have problems. If you have any ideas LMK.

Reply:
I meant this line is only a starting point. I'll play with it back home.